### PR TITLE
Tpuvm nightly tests

### PR DESCRIPTION
**Pull request by zcain117**
example manual run: [logs](https://pantheon.corp.google.com/logs/viewer?interval=NO_LIMIT&project=xl-ml-test&rif_reserved&minLogLevel=0&expandAll=false&timestamp=2021-06-21T22:09:38.195000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22xl-ml-test%22%0Aresource.labels.location%3D%22us-central1%22%0Aresource.labels.cluster_name%3D%22xl-ml-test-us-central1%22%0Aresource.labels.namespace_name%3D%22automated%22%0Aresource.labels.pod_name%3D%22pt-nightly-resnet50-mp-conv-v3-8-1vm-manual-9nlfn-rcbgc%22%0Aresource.labels.container_name%3D%22train%22&scrollTimestamp=2021-06-21T20:50:05.028792115Z&dateRangeUnbound=forwardInTime)

### CODE REVIEW

File: tests/pytorch/nightly/resnet50-mp.libsonnet [hunk 1]
```diff
@@ -93,6 +94,67 @@ local utils = import 'templates/utils.libsonnet';
...
+          --num_epochs=%(num_epochs)d \
+      ||| % self.paramsOverride,
+    ),
+    podTemplate+:: {
+      spec+: {
+        containerMap+: {
+          train+: {
+            resources+: {
+              requests: {
+                cpu: '1',
```
Comment (will-cromar) [inline tests/pytorch/nightly/resnet50-mp.libsonnet:124]: You can use these fields of BaseTest here: https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/master/templates/base.libsonnet#L60

File: tests/pytorch/nightly/resnet50-mp.libsonnet [hunk 2]
```diff
@@ -93,6 +94,67 @@ local utils = import 'templates/utils.libsonnet';
...
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  local functional_tpu_vm = common.Functional {
+    paramsOverride: {
+      setup_commands: common.tpu_vm_nightly_install,
```
Comment (will-cromar) [inline tests/pytorch/nightly/resnet50-mp.libsonnet:135]: Should setup_commands also be in resnet50_tpu_vm?

File: tests/pytorch/nightly/resnet50-mp.libsonnet (hunk 1 above)
Comment (will-cromar) [reply tests/pytorch/nightly/resnet50-mp.libsonnet:124]: On second thought, this probably does not do what you want it to. These cpu/memory requests will apply to the container that runs SSH to connect to the TPU VM, not the TPU VM itself

**Review by will-cromar — CHANGES_REQUESTED** [review]

File: tests/pytorch/nightly/resnet50-mp.libsonnet [hunk 3]
```diff
@@ -118,5 +180,7 @@ local utils = import 'templates/utils.libsonnet';
     resnet50_MP + v3_8 + functional + timeouts.Hours(2),
     resnet50_gpu + common.Functional + v100 + timeouts.Hours(2),
     resnet50_gpu + common.Functional + v100x4 + timeouts.Hours(1),
+    common.PyTorchTest + resnet50_tpu_vm + v3_8 + functional_tpu_vm + timeouts.Hours(2),
```
Comment (will-cromar) [inline tests/pytorch/nightly/resnet50-mp.libsonnet:183]: To be consistent with the other tests, you can do this:

`local resnet50_tpu_vm = common.PyTorchTest { ...`

`resnet50_tpu_vm + v3_8 + functional_tpu_vm + timeouts.Hours(2) + experimental.PyTorchTpuVmMixin`

Comment (zcain117) [reply tests/pytorch/nightly/resnet50-mp.libsonnet]: Good idea, switched

File: tests/pytorch/nightly/resnet50-mp.libsonnet (hunk 2 above)
Comment (zcain117) [reply tests/pytorch/nightly/resnet50-mp.libsonnet:136]: I added them to the `paramsOverride` object such that they'll throw an error if not provided. However the setup commands were there in the yamls when I ran the manual test

File: tests/pytorch/nightly/resnet50-mp.libsonnet (hunk 1 above)
Comment (zcain117) [reply tests/pytorch/nightly/resnet50-mp.libsonnet:125]: This is carryover from [this change](https://source.cloud.google.com/xl-ml-test/ml-testing-accelerators/+/41f4a3519dbaf53f1ea1f8d5e2bef72e2ab6ca1c) you made to the resnet50 TPUVM test. I assumed it was here to make sure we request a small (Easily available) machine to do the setup commands. Let me know if you want me to remove it

**Comment by zcain117** [comment]
started a new manual run with mixin updates, logs will be [here](https://pantheon.corp.google.com/logs/viewer?interval=NO_LIMIT&project=xl-ml-test&rif_reserved&minLogLevel=0&expandAll=false&timestamp=2021-06-22T00:20:42.637000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22xl-ml-test%22%0Aresource.labels.location%3D%22us-central1%22%0Aresource.labels.cluster_name%3D%22xl-ml-test-us-central1%22%0Aresource.labels.namespace_name%3D%22automated%22%0Aresource.labels.pod_name%3D%22pt-nightly-resnet50-mp-conv-v3-8-1vm-manual-9s6fx-fzv6j%22%0Aresource.labels.container_name%3D%22train%22) if TPU schedules successfully

File: tests/pytorch/nightly/resnet50-mp.libsonnet (hunk 1 above)
Comment (will-cromar) [reply tests/pytorch/nightly/resnet50-mp.libsonnet:125]: I remember now. The base class for this test has a large resource requirement for TPU Nodes, but I lowered it for TPU VM

**Review by will-cromar — APPROVED** [review]
